### PR TITLE
[ValueTracking] Support non-constant idx for `computeKnownFPClass` of `insertelement`

### PR DESCRIPTION
**Pull request by goldsteinn**
- **[ValueTracking] Add tests for non-constant idx for fpclass of `insertelement`; NFC**
- **[ValueTracking] Support non-constant idx for `computeKnownFPClass` of `insertelement`**

### CODE REVIEW

File: llvm/lib/Analysis/ValueTracking.cpp [hunk 1]
```diff
@@ -5355,14 +5355,17 @@ void computeKnownFPClass(const Value *V, const APInt &DemandedElts,
     const Value *Vec = Op->getOperand(0);
     const Value *Elt = Op->getOperand(1);
     auto *CIdx = dyn_cast<ConstantInt>(Op->getOperand(2));
-    // Early out if the index is non-constant or out-of-range.
     unsigned NumElts = DemandedElts.getBitWidth();
-    if (!CIdx || CIdx->getValue().uge(NumElts))
-      return;
+    APInt DemandedVecElts = DemandedElts;
+    bool NeedsElt = true;
+    // If we know the index we are inserting too, clear it from Vec check.
```
Comment (arsenm) [inline llvm/lib/Analysis/ValueTracking.cpp:5361]:
```suggestion
    // If we know the index we are inserting to, clear it from Vec check.
```

**Review by arsenm — APPROVED** [review]
I believe computeKnownBits has the same issue

**Comment by goldsteinn** [comment]
> I believe computeKnownBits has the same issue

Have a PR for that too: https://github.com/llvm/llvm-project/pull/87707

File: llvm/lib/Analysis/ValueTracking.cpp (hunk 1 above)
Comment (goldsteinn) [reply llvm/lib/Analysis/ValueTracking.cpp:5361]: Will update before pushing.

**Pull request closed without merging**